apps/aecore/test/aec_chain_tests.erl

```diff
@@ -1117,10 +1157,28 @@ fees_delayed_reward() ->
     DictBal3 = orddict:from_list(Balances3),
 
     %% Check rewards are granted for the first three key blocks, with fees of first two generations
-    ?assertEqual({ok, MiningReward1 + MiningReward2 + MiningReward3 + Fee1 + Fee2},
+    ?assertEqual({ok, split_reward(MiningReward1 + MiningReward2 + MiningReward3 + Fee1 + Fee2)},
                  orddict:find(PubKey3, DictBal3)),
     ok.
 
+
+%% HACK to avoid mocking of aec_dev_reward module - doing so will uncover race conditions
+%% between repeated setup and teardown functions (as the mocked module is global)
+with_dev_reward_env_instrumentation(KVs, F) ->
+    Prev = lists:foldl(
+             fun ({K, _}, Acc) ->
+                     case application:get_env(aecore, K) of
+                         {ok, PrevVal} -> [{set_env, [K, PrevVal]} | Acc];
+                         undefined -> [{unset_env, [K]} | Acc]
+                     end
+             end, [], KVs),
+    [application:set_env(aecore, K, V) || {K, V} <- KVs],
+    try F()
```

Comment:
I plan to come back to tests after other comments clarified - but from a quick look it seems to me that the set_env (and teardown) can be moved to proper setup/teardown.